File: 0118_pascals_triangle.py
# https://leetcode-cn.com/problems/pascals-triangle/

class Solution:
    def generate(self, numRows):
        """
        Inputs:
            numRows: int
        Outputs:
            res: list[list[int]]
        """
        
        res = []
        for i in range(numRows):
            if i == 0:
                res.append([1])
            elif i == 1:
                res.append([1,1])
            else:
                row = res[-1]
                newRow = list(range(1,i+2))
                for j in range(1,i):
                    newRow[j] = row[j-1] + row[j]
                newRow[-1] = 1
                res.append(newRow)
        return res

File: Cpp/0118_pascals_triangle.cpp
// https://leetcode.cn/problems/pascals-triangle/

class Solution {
public:
    vector<vector<int>> generate(int numRows) {
        vector<vector<int>> res;
        if (numRows == 1) {
            res = {{1}};
            return res;
        }
        else if (numRows == 2) {
            res = {{1}, {1,1}};
            return res;
        }
        else {
            res = generate(numRows-1);
            vector<int> row(numRows, 1);
            for (int i = 1; i < numRows-1; ++i) {
                row[i] = res[numRows-2][i-1] + res[numRows-2][i];
            }
            res.push_back(row);
            return res;
        }
    }
};
